test/EmbedIO.Tests/Issues/Issue_PassHttpException.cs

```diff
@@ -0,0 +1,30 @@
+﻿using EmbedIO.Tests.TestObjects;
+using EmbedIO.WebApi;
+using NUnit.Framework;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace EmbedIO.Tests.Issues
+{
+    [TestFixture]
+    public class Issue_PassHttpException : EndToEndFixtureBase
+    {
+        protected override void OnSetUp()
+        {
+            Server.WithWebApi("/api", o =>
+                {
+                    o.WithController<TestController>();
+                    o.OnUnhandledException = (ctx, ex) => throw HttpException.BadRequest();
```

Comment:
Maybe, just to be sure, use some really weird status code such as `417 I'm a teapot` or something.

Reply:
Usually, this will return a 404.